datamover/pkg/drivers/lifecycle/incloud_transition.go

```diff
@@ -51,7 +51,10 @@ func doInCloudTransition(acReq *datamover.LifecycleActionRequest) error {
 		return errors.New(DMERR_TransitionInprogress)
 	}
 
-	ctx, _ := context.WithTimeout(context.Background(), CLOUD_OPR_TIMEOUT*time.Second)
+	// min is 1 minute, max is 30 days, default is 1 hour, user defined value should not less than min and not more than
+	// max, otherwise default will be used
+	tmout := migration.GetCtxTimeout("OBJECT_MOVE_TIME", SECONDS_ONE_MINUTE, SECONDS_30_DAYS, SECONDS_ONE_HOUR)
+	ctx, _ := context.WithTimeout(context.Background(), tmout)
```

Comment:
my concern is this...recovering from a timed out timeout is usually messy. Further, arriving at an optimal timeout is also not easy. Should we do something like an auto-extend-timeout option ? How it will work is 

1. say you have a timeout of 30 seconds
2. 5 files need to be moved from A to B
3. at run time, we know 3 files have been moved and 20 seconds have elapsed (we will run out of time)
4. now, we auto-extend the timeout by 15 more seconds (to allow task to finish) if the user had selected this auto-extend option

Reply:
Thanks for your suggestion, for auto-extend-timeout, I think we can do it as an enhancement in the future. Currently, for migration, if it failed, user can run it again. For lifecycle management, it will try again in the next schedule round automatically.

Reply:
ok